server.js

```diff
@@ -1,23 +1,187 @@
 const express = require("express");
 const cors = require("cors");
+//const { check, validationResult } = require("express-validator");
 
 const app = express();
-
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
 app.use(cors());
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
-  text: "Welcome to CYF chat system!",
+  text: "Welcome to CYF chat system!"
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [  welcomeMessage,
+  {
+    id: 1,
+    from: "Joanna",
+    text: "Hey, how are you?"
+  },
+  {
+    id: 2,
+    from: "Tas",
+    text: "Are you going us today"
+  },
+  
+  {
+    id: 3,
+    from: "Altom",
+    text: "Are you going us today"
+  }
+  
+];
 
-app.get("/", function (request, response) {
+app.get("/", function(request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// const randomId = (){
+//   return Math.floor(Math.random() * 1000000000);
+// }
+
+//read  all messages
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+// create message
+app.post("/messages", (req, res) => {
+  let today = new Date();
+  let date =
+    today.getFullYear() + "-" + (today.getMonth() + 1) + "-" + today.getDate();
+  let time =
+    today.getHours() + 1 + ":" + today.getMinutes() + ":" + today.getSeconds();
+  let dateTime = date + " " + time;
+  let newId = messages.length + 1;
+  if (!req.body.from || !req.body.text) {
+    res.status(422).json("Field cannot be empty ");
+  } else {
+    let newPost = {
+      timeSent: dateTime,
+      id: newId,
+      from: req.body.from,
+      text: req.body.text
+    };
+    messages.push(newPost);
+    res.json(messages);
+  }
+});
+
+//read one message specified by an ID
+
+// app.get("/messages/:id", (req, res) => {
+//   let id = req.params.id;
+//   let found = messages.find(message => message.id == id);
+//   console.log(found);
+//   res.json(found);
+// });
```

Comment:
comments can be really useful to describe code and to help you during development, but it is good practice to delete commented out code once it is no longer needed.